From 9eaf1fc6efde25f4094e0bf787772901a874aa0a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 27 Feb 2025 22:34:21 +0800
Subject: [PATCH 2/2] FROMLIST: quirks: raise AttrTrackpointMultiplier for
 ThinkPad T470/T480/A485 to 0.75

As I tested my friend's ThinkPad A485, I found that the default multiplier
resulted in jumpy cursor and a slightly too quick acceleration curve. Upon
checking for Lenovo quirks, I found that since commits 383a60abea8e
("Better Thinkpad T480 trackpoint multiplier") and a1566e3492ae ("quirks:
Thinkpad T470 trackpoint multiplier"), the TrackPoint multiplier for both
T470 and T480 (which shares the same keyboard FRUs with the A485) were set
to 0.4.

However, per my testing, by setting the multiplier to 0.4, the TrackPoint
speed became so painfully slow that it began to hurt my index finger...
I suspect the original commiters have set custom acceleration curves on
their own system, but I might be wrong.

Playing with the multiplier, I found 0.75 to be the most appropriate and,
interestingly, with anything >= 0.8, the TrackPoint began to become jumpy,
with the cursor appearing to have a very low poll rate on the screen (the
higher the multiplier, the worse it gets).

Since, as I mentioned above, the keyboard and TrackPoint parts are shared
between these three models, I'm assuming that this multiplier will work
well for them.

Link: https://gitlab.freedesktop.org/libinput/libinput/-/merge_requests/1149
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 quirks/50-system-lenovo.quirks | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/quirks/50-system-lenovo.quirks b/quirks/50-system-lenovo.quirks
index 404efa20..6ebdd21f 100644
--- a/quirks/50-system-lenovo.quirks
+++ b/quirks/50-system-lenovo.quirks
@@ -23,12 +23,17 @@ ModelLenovoT450Touchpad=1
 [Lenovo T470 Trackpoint]
 MatchName=*TPPS/2 IBM TrackPoint
 MatchDMIModalias=dmi:*svnLENOVO:*:pvrThinkPadT470:*
-AttrTrackpointMultiplier=0.4
+AttrTrackpointMultiplier=0.75
 
 [Lenovo T480 Trackpoint]
 MatchName=*TPPS/2 IBM TrackPoint
 MatchDMIModalias=dmi:*svnLENOVO:*:pvrThinkPadT480:*
-AttrTrackpointMultiplier=0.4
+AttrTrackpointMultiplier=0.75
+
+[Lenovo A485 Trackpoint]
+MatchName=*TPPS/2 IBM TrackPoint
+MatchDMIModalias=dmi:*svnLENOVO:*:pvrThinkPadA485:*
+AttrTrackpointMultiplier=0.75
 
 # Touchpad is a clickpad but INPUT_PROP_BUTTONPAD is not set, see
 # https://gitlab.freedesktop.org/libinput/libinput/issues/177
-- 
2.48.1

